lib/ansible/modules/database/postgresql/pg_hba.py

```diff
@@ -0,0 +1,769 @@
+        if contype not in PgHbaTypes:
+            raise PgHbaError("invalid connection type {0} (should be one of '{1}').".format(contype, "', '".join(PgHbaTypes)))
+        # Add the job
+        rule = dict(zip(PgHbaHDR, [contype, databases, users, source, netmask, method, options]))
+
+        for db in databases.split(','):
+            for usr in users.split(','):
+
+                rule = dict(zip(PgHbaHDR, [contype, db, usr, source, netmask, method, options]))
+
+                if contype == 'local':
+                    del rule['src']
+                    del rule['mask']
+                elif '/' in source:
+                    del rule['mask']
+                elif ipv4_re.search(source):
+                    if not netmask:
+                        rule['src'] += '/32'
+                elif '/' in source:
+                    if not netmask:
+                        rule['src'] += '/128'
+                else:
+                    del rule['mask']
+
+                self.cleanEmptyRuleKeys(rule)
+
+                line = [rule[k] for k in PgHbaHDR if k in rule.keys()]
+                rule['line'] = "\t".join(line)
+                yield rule
+
+    def add_rule(self, rule):
+        key = self.rule2key(rule)
+        try:
+            oldrule = self.rules[key]
+            ekeys = set(oldrule.keys() + rule.keys())
+            ekeys.remove('line')
+            for k in ekeys:
+                if oldrule[k] != rule[k]:
+                    raise Exception('')
+        except:
+            self.rules[key] = rule
+            self.diff['after'] += rule['line'] + '\n'
+            if rule['db'] not in ['all', 'samerole', 'samegroup', 'replication']:
+                databases = set(rule['db'].split(','))
+                self.databases.update(databases)
+            if rule['usr'] != 'all':
+                user = rule['usr']
+                if user[0] == '+':
+                    user = user[1:]
+                self.users.add(user)
+
+    def remove_rule(self, rule):
+        keys = self.rule2key(rule)
+        try:
+            del self.rules[keys]
+            self.diff['before'] += rule['line'] + '\n'
+        except:
+            pass
+
+    def get_rules(self):
+        ret = []
+        for k in self.rules.keys():
+            rule = self.rules[k]
+            del rule['line']
+            ret.append(rule)
+        return ret
+
+    def render(self):
+        comment = '\n'.join(self.comment)
+        sorted_rules = sorted(self.rules.values(), key=self.rule2weight)
+        rule_lines = '\n'.join([r['line'] for r in sorted_rules])
+        result = comment + '\n' + rule_lines
+        # End it properly with a linefeed (if not already).
+        if result and result[-1] not in ['\n', '\r']:
+            result += '\n'
+        return result
+
+    def changed(self):
+        if self.diff['before'] or self.diff['after']:
```

Comment:
this could be `return self.diff['before'] or self.diff['after']`

Reply:
fixed